spec/app/domain/conjur/validate_trusted_proxies_spec.rb

```diff
@@ -0,0 +1,297 @@
+                                                  valid_proxy_ip_2,]}
+
+
+  let(:mocked_fetch_trusted_proxy_empty_list) {double("EmptyTrustedProxiesList")}
+  let(:mocked_fetch_trusted_proxy_returns_error) {double("EmptyTrustedProxiesList")}
+
+  let(:mocked_fetch_trusted_proxy_returns_1_cidr_32) {double("TrustedProxiesListContains1CIDR32")}
+  let(:mocked_fetch_trusted_proxy_returns_1_cidr_24) {double("TrustedProxiesListContains1CIDR24")}
+  let(:mocked_fetch_trusted_proxy_returns_1_cidr_8) {double("TrustedProxiesListContains1CIDR8")}
+  let(:mocked_fetch_trusted_proxy_returns_3_ranges) {double("TrustedProxiesListContains3Ranges")}
+  let(:mocked_fetch_trusted_proxy_returns_5_ranges) {double("TrustedProxiesListContains3Ranges")}
+
+  before(:each) do
+    allow(mocked_fetch_trusted_proxy_empty_list).to receive(:call)
+                                                        .with({:account => valid_account})
+                                                        .and_return(empty_trusted_proxy_list)
+
+    allow(mocked_fetch_trusted_proxy_returns_error).to receive(:call)
+                                                           .with({:account => valid_account})
+                                                           .and_raise(::Errors::Conjur::TrustedProxiesFetchFailed)
+
+    allow(mocked_fetch_trusted_proxy_returns_error).to receive(:call)
+                                                           .with({:account => valid_account})
+                                                           .and_raise(::Errors::Conjur::TrustedProxiesFetchFailed)
+
+    allow(mocked_fetch_trusted_proxy_returns_1_cidr_32).to receive(:call)
+                                                               .with({:account => valid_account})
+                                                               .and_return(trusted_proxies_list_1_cidr_32)
+    allow(mocked_fetch_trusted_proxy_returns_1_cidr_24).to receive(:call)
+                                                               .with({:account => valid_account})
+                                                               .and_return(trusted_proxies_list_1_cidr_24)
+    allow(mocked_fetch_trusted_proxy_returns_1_cidr_8).to receive(:call)
+                                                              .with({:account => valid_account})
+                                                              .and_return(trusted_proxies_list_1_cidr_8)
+
+    allow(mocked_fetch_trusted_proxy_returns_3_ranges).to receive(:call)
+                                                              .with({:account => valid_account})
+                                                              .and_return(trusted_proxies_list_3_ranges)
+  end
+  #  ____  _   _  ____    ____  ____  ___  ____  ___
+  # (_  _)( )_( )( ___)  (_  _)( ___)/ __)(_  _)/ __)
+  #   )(   ) _ (  )__)     )(   )__) \__ \  )(  \__ \
+  #  (__) (_) (_)(____)   (__) (____)(___/ (__) (___/
+
```

Comment:
@eladkug can you review the following tests please ?

Reply:
cool!